lib/http_clients/b2b.ex

```diff
@@ -0,0 +1,51 @@
+defmodule HttpClients.B2b do
+  @moduledoc """
+  Client for B2b calls
+  """
+  alias HttpClients.B2b.Indication
+
+  @spec create_indication(Tesla.Client.t(), String.t()) :: {:error, any} | {:ok, Tesla.Env.t()}
+  def create_indication(%Tesla.Client{} = client, indication) do
+    case Tesla.post(client, "/v2/indications", indication) do
+      {:ok, %Tesla.Env{status: 201} = response} -> {:ok, build_indication(response.body["data"])}
+      {:ok, %Tesla.Env{} = response} -> {:error, response}
```

Comment:
```suggestion
      {:ok, %Tesla.Env{} = response} -> {:error, response}
      {:error, reason} -> {:error, reason}
```
Acho que está faltando teste para essas duas situações?

Reply:
@mateusmalvarcreditas eu vi que você fez o merge do que eu coloquei acima, mas idéia não era essa e sim um questionamento sobre faltar testes para aquelas situações de retorno de erro.

Reply:
Boa, não tinha visto a mensagem. vou colocar alguns testes aqui.

Reply:
@adrianosiq coloquei um erro 500 no test, pensei em colocar mais, mas teoricamente esperamos o restante nos serviços. Seria isso mesmo?

Reply:
```suggestion
test "returns error when service is unavailable" do
   mock(fn %{method: :post, url: @indications_url} -> {:error, :timeout} end)
   assert B2b.create_indication(client(), proponent_id) == {:error, :timeout}
end

test "returns error when mobile_phone is invalid" do
    response_body = %{"errors" => %{"mobile_phone" => ["is invalid"]}}
    mock(fn %{method: :post, url: @indications_url} -> json(response_body, status: 422) end)
    assert {:error, expected_response} = B2b.create_indication(client(), @indication)
    assert %Tesla.Env{body: ^response_body, status: 422} = expected_response
end
```

Talvez? O primeiro cobre o `{:error, reason} -> {:error, reason}` e o segundo {:ok, %Tesla.Env{} = response} -> {:error, response}